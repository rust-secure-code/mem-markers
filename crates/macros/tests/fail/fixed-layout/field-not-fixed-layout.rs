use mem_markers::FixedLayout;

#[derive(FixedLayout)]
#[repr(C)]
struct Foo {
    a: Other,
}

struct Other {}
fn main() {}
