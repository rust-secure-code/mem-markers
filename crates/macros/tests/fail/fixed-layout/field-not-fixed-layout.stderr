error[E0277]: the trait bound `Other: mem_markers::fixed_layout::FixedLayout` is not satisfied
 --> $DIR/field-not-fixed-layout.rs:6:8
  |
3 | #[derive(FixedLayout)]
  |          ----------- required by this bound in `FooImplsFixedLayout`
4 | #[repr(C)]
5 | struct Foo {
  |        ---
6 |     a: Other,
  |        ^^^^^ the trait `mem_markers::fixed_layout::FixedLayout` is not implemented for `Other`
